### PR TITLE
Fixes from PHPStan

### PR DESCRIPTION
**Pull request by carusogabriel**
Some fixes from [`PHPStan`](https://github.com/phpstan/phpstan) levels 0, 1 and 2 :man_technologist:

### CODE REVIEW

File: src/Psy/Input/FilterOptions.php [hunk 1]
```diff
@@ -135,7 +135,7 @@ private function validateRegex($pattern)
         set_error_handler(['Psy\Exception\ErrorException', 'throwException']);
         try {
             preg_match($pattern, '');
-        } catch (ErrorException $e) {
+        } catch (\ErrorException $e) {
```
Comment (bobthecow) [inline src/Psy/Input/FilterOptions.php]: i think this one is supposed to be `Psy\Exception\ErrorException`, so we're missing a `use` statement at the top, not a slash here.

File: src/Psy/TabCompletion/Matcher/AbstractContextAwareMatcher.php [hunk 2]
```diff
@@ -44,7 +44,7 @@ public function setContext(Context $context)
     /**
      * Get a Context variable by name.
      *
-     * @param $var Variable name
+     * @param Variable $var name
```
Comment (bobthecow) [inline src/Psy/TabCompletion/Matcher/AbstractContextAwareMatcher.php]: this should be `@param string $var Variable name`

File: test/Psy/Test/TabCompletion/AutoCompleterTest.php [hunk 3]
```diff
@@ -21,9 +21,9 @@
 class AutoCompleterTest extends \PHPUnit\Framework\TestCase
 {
     /**
-     * @param $line
-     * @param $mustContain
-     * @param $mustNotContain
+     * @param string $line
+     * @param array  $mustContain
+     * @param array  $mustNotContain
```
Comment (bobthecow) [inline test/Psy/Test/TabCompletion/AutoCompleterTest.php]: see, this is why i don't document tests :P

Comment (theofidry) [reply test/Psy/Test/TabCompletion/AutoCompleterTest.php]: also `array` is not very useful here... Just bump to 7.1 and use typehints 😄

Comment (carusogabriel) [reply test/Psy/Test/TabCompletion/AutoCompleterTest.php]: @bobthecow I don't document they either, but these ones with `dataProvider`s, are important :)

@theofidry Bump to 7.1 is out of the question, I guess :grin:

**Comment by bobthecow** [comment]
Thanks again :)

**Pull request merged**